packages/mobile/src/components/TotalLineItem.v2.tsx

```diff
@@ -11,16 +11,23 @@ import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
 import { Namespaces } from 'src/i18n'
 import { LocalCurrencyCode } from 'src/localCurrency/consts'
 import { useExchangeRate, useLocalCurrencyCode } from 'src/localCurrency/hooks'
+import { CurrencyInfo } from 'src/send/SendConfirmation'
 
 interface Props {
   title?: string
   amount: MoneyAmount
   hideSign?: boolean
+  currencyInfo?: CurrencyInfo
 }
 
-export default function TotalLineItem({ title, amount, hideSign }: Props) {
-  const localCurrencyCode = useLocalCurrencyCode()
-  const localCurrencyExchangeRate = useExchangeRate()
+export default function TotalLineItem({ title, amount, hideSign, currencyInfo }: Props) {
+  let localCurrencyCode = useLocalCurrencyCode()
+  let localCurrencyExchangeRate = useExchangeRate()
+  if (currencyInfo) {
+    localCurrencyCode = currencyInfo.currencyCode
+    localCurrencyExchangeRate = currencyInfo.exchangeRate
+  }
+
```

Comment:
What do you think about simplifying the code a bit and defining these variables by using a ternary like you did elsewhere?

Reply:
I tried doing that but it won't let me put React hooks in conditional operators :( I also hate this section of code and am totally open to other workarounds